tests/db_functions/datetime/test_extract_trunc.py

```diff
@@ -1935,3 +1935,113 @@ def test_trunc_invalid_field_with_timezone(self):
             DTModel.objects.annotate(
                 hour_melb=Trunc("start_time", "hour", tzinfo=melb),
             ).get()
+
+    def test_trunc_with_etc_offset_timezone(self):
+        start_datetime = datetime(2015, 6, 15, 14, 30, 50, 321)
+        end_datetime = datetime(2016, 6, 15, 14, 10, 50, 123)
+        start_datetime = timezone.make_aware(start_datetime)
+        end_datetime = timezone.make_aware(end_datetime)
+        self.create_model(start_datetime, end_datetime)
+        self.create_model(end_datetime, start_datetime)
+        etc_timezone = zoneinfo.ZoneInfo("Etc/GMT+10")
+
+        def test_datetime_kind(kind):
+            self.assertQuerySetEqual(
+                DTModel.objects.annotate(
+                    truncated=Trunc(
+                        "start_datetime",
+                        kind,
+                        output_field=DateTimeField(),
+                        tzinfo=etc_timezone,
+                    )
+                ).order_by("start_datetime"),
+                [
+                    (
+                        start_datetime,
+                        truncate_to(
+                            start_datetime.astimezone(etc_timezone), kind, etc_timezone
+                        ),
+                    ),
+                    (
+                        end_datetime,
+                        truncate_to(
+                            end_datetime.astimezone(etc_timezone), kind, etc_timezone
+                        ),
+                    ),
+                ],
+                lambda m: (m.start_datetime, m.truncated),
+            )
```

Comment:
I think it's more readable to pull out some variables here. Something like:

```suggestion
            truncated_start = truncate_to(
                start_datetime.astimezone(etc_timezone), kind, etc_timezone
            )
            truncated_end = truncate_to(
                end_datetime.astimezone(etc_timezone), kind, etc_timezone
            )
            queryset = DTModel.objects.annotate(
                truncated=Trunc(
                    "start_datetime",
                    kind,
                    output_field=DateTimeField(),
                    tzinfo=etc_timezone,
                )
            ).order_by("start_datetime")
            self.assertQuerySetEqual(
                queryset,
                [
                    (start_datetime, truncated_start),
                    (end_datetime, truncated_end),
                ],
                lambda m: (m.start_datetime, m.truncated),
            )
```

 You can also avoid the `lambda` in `assertQuerySetEqual` by using `values_list` instead:

```suggestion
            truncated_start = truncate_to(
                start_datetime.astimezone(etc_timezone), kind, etc_timezone
            )
            truncated_end = truncate_to(
                end_datetime.astimezone(etc_timezone), kind, etc_timezone
            )
            queryset = DTModel.objects.annotate(
                truncated=Trunc(
                    "start_datetime",
                    kind,
                    output_field=DateTimeField(),
                    tzinfo=etc_timezone,
                )
            ).order_by("start_datetime")
            self.assertSequenceEqual(
                queryset.values_list("start_datetime", "truncated"),
                [
                    (start_datetime, truncated_start),
                    (end_datetime, truncated_end),
                ],
            )
```